### PR TITLE
Add indexer setup to the infrastructure scripts

### PR DESCRIPTION
**Pull request by ratik21**

### CODE REVIEW

File: infrastructure/README.md [hunk 1]
```diff
@@ -131,3 +131,51 @@ KMD is not started by default (with `make start-private`). You need to run it se
...
+For this we provide `make setup-postgresql` command. Running it will [install](https://www.postgresql.org/download/linux/ubuntu/) postgresql database on your system and setup a new user & database. Config vars are provided below:
+```
+host=localhost
+port=5432
+username=algorand
+dbname=pgdb
+password=indexer
+```
+
+Please note that this installation is currently compatible with `linux based` distributions only. For other OS, try the installation from [source](https://www.postgresql.org/download/).
```
Comment (robert-zaremba) [inline infrastructure/README.md:167]:
```suggestion
Please note that this installation is currently compatible with Linux based distributions only. For other OS, check [postresql documentation](https://www.postgresql.org/download/).
```

File: infrastructure/README.md [hunk 2]
```diff
@@ -131,3 +131,51 @@ KMD is not started by default (with `make start-private`). You need to run it se
...
+
+#### Step2: Start Indexer
+
+**Note:** Make sure you have setup a private network up and running (using `make setup-private-net`). For more instructions check [setting up a local node](./#setting-up-a-local-node) section.
```
Comment (robert-zaremba) [inline infrastructure/README.md:171]:
```suggestion
**Note:** Make sure you have setup a private network up and running (using `make setup-private-net` or Sandbox). For more instructions check [setting up a local node](./#setting-up-a-local-node) section.
```

File: infrastructure/Makefile [hunk 3]
```diff
@@ -91,4 +101,54 @@ remove-reach:
...
+	@sudo -u postgres psql -c "grant all privileges on database $(POSTGRES_DBNAME) to $(POSTGRES_UNAME)"
+# List databases
+	@sudo -u postgres psql -c "\l"
+
+
+# Sets up indexer in local mode (to be used with private-net)
+setup-indexer:
+	@if ! [ -d ./node_data ] ;  then echo "./node_data directory does not exist - please create a new private network first" ; false; fi
+# https://developer.algorand.org/docs/run-a-node/setup/indexer/#extract-the-binaries-to-a-specific-directory
+	mkdir -p ~/indexer
```
Comment (robert-zaremba) [inline infrastructure/Makefile:144]: aren't we using `~/.algorand-indexer`?

File: infrastructure/README.md [hunk 4]
```diff
@@ -131,3 +131,51 @@ KMD is not started by default (with `make start-private`). You need to run it se
...
+
+After setting up the database, you can use `make setup-indexer` to add local indexer binary (downloaded in `~/indexer`) and start the indexer by connecting to database and your local algod node.
```
Comment (robert-zaremba) [inline infrastructure/README.md:173]: aren't we using `.algorand-indexer`? If not, shall we use a directory with algorand in name? Eg: ~/.algorand-indexer-download

File: infrastructure/Makefile (hunk 3 above)
Comment (ratik21) [reply infrastructure/Makefile:144]: Actually the binary executable file is also named as `algorand-indexer`. So for local based setup i kept it as `~/indexer`

File: infrastructure/README.md (hunk 4 above)
Comment (ratik21) [reply infrastructure/README.md:173]: `~/.algorand-indexer-download` works. Updating.

File: infrastructure/Makefile [hunk 5]
```diff
@@ -91,4 +101,55 @@ remove-reach:
...
+# log connection info
+	sudo -u postgres psql -c "\conninfo"
+# Create user (with password) and database. Grant permissions.
+	@echo "Setting up database: User=$(POSTGRES_UNAME) password=$(POSTGRES_PASS) dbname=$(POSTGRES_DBNAME)"
+	@sudo -u postgres createuser $(POSTGRES_UNAME)
+	@sudo -u postgres createdb $(POSTGRES_DBNAME)
+	@sudo -u postgres psql -c "alter user $(POSTGRES_UNAME) with encrypted password '$(POSTGRES_PASS)';"
+	@sudo -u postgres psql -c "grant all privileges on database $(POSTGRES_DBNAME) to $(POSTGRES_UNAME)"
+# List databases
+	@sudo -u postgres psql -c "\l"
```
Comment (robert-zaremba) [inline infrastructure/Makefile:137]: let's also add docker based postgres setup and update the documentation.

**Review by robert-zaremba — APPROVED** [review]
pre-approving. Let's add docker based postgres setup as well.

File: infrastructure/Makefile (hunk 5 above)
Comment (ratik21) [reply infrastructure/Makefile:137]: docker based setup already has the postgres (indexer-db) container: [https://github.com/algorand/indexer/blob/develop/docker-compose.yml#L17](https://github.com/algorand/indexer/blob/develop/docker-compose.yml#L17)

**Review by amityadav0 — APPROVED** [review]

File: infrastructure/Makefile (hunk 5 above)
Comment (ratik21) [reply infrastructure/Makefile:140]: updating docs

**Pull request merged**